crates/bevy_ecs/src/schedule/condition.rs

```diff
@@ -946,6 +946,24 @@ pub mod common_conditions {
         move |mut removals: RemovedComponents<T>| removals.iter().count() != 0
     }
 
+    /// Generates a [`Condition`](super::Condition)-satisfying closure that
+    /// returns `true` if there are any entities with a component of the given
+    /// type changed. Newly added components are considered changed.
+    pub fn any_component_changed<T: Component>() -> impl FnMut(Query<(), Changed<T>>) -> bool + Clone
+    {
+        move |query: Query<(), Changed<T>>| !query.is_empty()
+    }
+
+    /// Generates a [`Condition`](super::Condition)-satisfying closure that
+    /// returns `true` if there are any entities with a component of the given
+    /// type changed or removed. Newly added components are considered changed.
+    pub fn any_component_changed_or_removed<T: Component>(
+    ) -> impl FnMut(Query<(), Changed<T>>, RemovedComponents<T>) -> bool + Clone {
+        move |query: Query<(), Changed<T>>, mut removals: RemovedComponents<T>| {
+            !query.is_empty() || removals.iter().count() != 0
```

Comment:
```suggestion
            // NOTE: The check for removed components must come before the check for changed compoents.
            // If `query.is_empty()` is called first, then the short-circuiting behavior means that the removed
            // components buffer may not be consumed every frame.
            removals.iter().count() != 0 || !query.is_empty()
```

Ordering it in this way is also slightly faster, since checking for removed components doesn't do any iteration.